### ISSUE TITLE
Likes are not supported in this item

### ISSUE THREAD

**Issue by Saintenr** [pnp/pnpjs#2325]
### Category
- [ ] Enhancement
- [x] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [   "@pnp/sp": "^3.4.1"      ]

Please specify what version(s) of SharePoint you are targeting: [   SP Online      ]

### Expected / Desired Behavior / Question

I want to read the number of likes of an item from the document library

### Observed Behavior

I would like to link or unlink an item in a document library. Unfortunately, I get an error when reading the likes that the item does not support liking. However, the liken or disliken works.

![social code](https://user-images.githubusercontent.com/33520976/174590809-19c80473-3764-497d-98bf-c979dbd124e2.png)

I get this error when i run getLikeStatus() function
![error like](https://user-images.githubusercontent.com/33520976/174590510-5aa944a9-a7ae-432a-8500-f6e1bb3128dc.png)

Like and dislike function is working:
![like on item](https://user-images.githubusercontent.com/33520976/174590530-54c2e3e9-4aef-43bd-9d1a-b5f0508f8cc0.png)

My document library settings:
![rating settings](https://user-images.githubusercontent.com/33520976/174591961-382d3638-3876-409c-8b31-d5876b2420e0.png)

### Steps to Reproduce

The item is found via the sharepoint search, then the item is loaded via the unique id. See code in the upper section

Thanks for your help

**Comment by Saintenr** [comment]
LikedByInformation is working, but didn't show like count

**Comment by juliemturner** [comment]
Well... since this is the second time someone asks about likes in a list/library (also something similar -- liking comments on list items -- reported in #2313 by [chaitanya-SharePoint](https://github.com/chaitanya-SharePoint)) I figured I'd ask the experts in UI stuff at [@Sympraxis](https://github.com/SympraxisConsulting).

The consensus is that liking documents and list items is only available in modern SharePoint Team sites and NOT in modern Communication sites. Don't ask me why, but that's the deal.  You can see that there's a list/library settings called Rating settings that enables this feature. I suspect that if you're having issues it's because you a. are not in a team site, or b. you haven't enabled ratings.  This is not necessarily the answer for liking comments but it's a good direction to look in.

**Comment by Saintenr** [comment]
> Well... since this is the second time someone asks about likes in a list/library (also something similar -- liking comments on list items -- reported in #2313 by [chaitanya-SharePoint](https://github.com/chaitanya-SharePoint)) I figured I'd ask the experts in UI stuff at [@Sympraxis](https://github.com/SympraxisConsulting).
> 
> The consensus is that liking documents and list items is only available in modern SharePoint Team sites and NOT in modern Communication sites. Don't ask me why, but that's the deal. You can see that there's a list/library settings called Rating settings that enables this feature. I suspect that if you're having issues it's because you a. are not in a team site, or b. you haven't enabled ratings. This is not necessarily the answer for liking comments but it's a good direction to look in.

Thanks for your reply. The rating function is active in this list and i was wondering that like and dislike is working without a problem. Only read back the likes on the item is not working and i was wondering that number of likes and likes count are different columns and at the like/dislike function "Number of likes" will be filled and at the getLikedByInformation function it gets these three fields:

![getLikedByInformation](https://user-images.githubusercontent.com/33520976/175210510-277a5414-90a1-48da-87b4-5392a9562b01.png)

But not the Number of likes and not that i liked the item already.

I am not sure what site type i'am using... I didn't set this up on my own.

**Comment by juliemturner** [comment]
You can tell what type of site it is by weather or not it has left hand navigation... if it does it's a team site.

In the "teach a person to fish" realm I will tell you how I would go about getting your answers... 

Either use PnPjs or the browser and get the fields of the list... usually for speed I use a browser, paste the following url into the window and then paste the results into vscode into a file with an xml ext and then format the document and look for the fields that I want to use, find their internal names, and query them directly in the select for getting the items... 

https://contoso.sharepoint.com/sites/myteamsite/_api/web/Lists/GetByTitle('pnplist')/fields

Long story short, a really quick look shows that LikedBy is a Person field which would theoretically mean it would be expandable... so working off that... the following worked fine for me:

```ts
const w = await sp.web.lists.getByTitle("pnplist").items.select("Id", "Title", "LikesCount", "LikedBy/Id", "LikedBy/EMail").expand("LikedBy")();
```

Which gives me the numbe of likes, and an array of the Id and Email address of the users who liked it...

**Comment by patrick-rodgers** [comment]
Closing this issue as answered. If you have additional questions or we did not answer your question, please open a new issue, ref this issue, and provide any additional details available. Thank you!
